libs/vertexai/langchain_google_vertexai/vectorstores/vectorstores.py

```diff
@@ -0,0 +1,258 @@
+        query: str,
+        k: int = 4,
+        filter: Optional[List[Namespace]] = None,
+        **kwargs: Any,
+    ) -> List[Document]:
+        """Return docs most similar to query.
+        Args:
+            query: The string that will be used to search for similar documents.
+            k: The amount of neighbors that will be retrieved.
+            filter: Optional. A list of Namespaces for filtering the matching results.
+                For example:
+                [Namespace("color", ["red"], []), Namespace("shape", [], ["squared"])]
+                will match datapoints that satisfy "red color" but not include
+                datapoints with "squared shape". Please refer to
+                https://cloud.google.com/vertex-ai/docs/matching-engine/filtering#json
+                 for more detail.
+        Returns:
+            A list of k matching documents.
+        """
+        return [
+            document
+            for document, _ in self.similarity_search_with_score(query, k, filter)
+        ]
+
+    def add_texts(
+        self,
+        texts: Iterable[str],
+        metadatas: Union[List[dict], None] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Run more texts through the embeddings and add to the vectorstore.
+        Args:
+            texts: Iterable of strings to add to the vectorstore.
+            metadatas: Optional list of metadatas associated with the texts.
+            kwargs: vectorstore specific parameters.
+        Returns:
+            List of ids from adding the texts into the vectorstore.
+        """
+
+        # Makes sure is a list an can get the length, should we support iterables?
+        # metadata is a list so probably not?
+        texts = list(texts)
+        ids = self._generate_unique_ids(len(texts))
+
+        self._document_storage.batch_store_by_id(ids=ids, texts=texts)
+        embeddings = self._embeddings.embed_documents(texts)
+        self._searcher.add_to_index(ids, embeddings, metadatas, **kwargs)
+
+        return ids
+
+    @classmethod
+    def from_texts(
+        cls: Type["_BaseVertexAIVectorStore"],
+        texts: List[str],
+        embedding: Embeddings,
+        metadatas: Union[List[dict], None] = None,
+        **kwargs: Any,
+    ) -> "_BaseVertexAIVectorStore":
+        """Use from components instead."""
+        raise NotImplementedError(
+            "This method is not implemented. Instead, you should initialize the class"
+            " with `VertexAIVectorSearch.from_components(...)` and then call "
+            "`add_texts`"
+        )
+
+    @classmethod
+    def _get_default_embeddings(cls) -> "TensorflowHubEmbeddings":
+        """This function returns the default embedding.
+        Returns:
+            Default TensorflowHubEmbeddings to use.
+        """
+
+        # TODO: Change to vertexai embbedingss
+
+        from langchain_community.embeddings import TensorflowHubEmbeddings
+
+        return TensorflowHubEmbeddings()
```

Comment:
should we maybe add a deprecation warning and switch to different default embeddings (VertexAI) so that we can drop dependency on `langchain_community` completely?